santorini-common/src/phase.rs

```diff
@@ -1,7 +1,23 @@
 use crate::objects::{player::Player, state::State};
+use crate::{command::Command, error::SantoriniError};
 
 #[derive(Debug)]
 pub enum Phase {
     InProgress(State),
     Win(State, Player),
 }
+
+impl Phase {
+    fn apply(self, command: &dyn Command) -> Result<Phase, SantoriniError> {
```

Comment:
Is the goal to eventually use this function in a fold or a try_fold?

Reply:
See integration tests.